### PR TITLE
Adds documentation regarding thread safety of `CompilerPipeline`

### PR DESCRIPTION
**Pull request by alancai98**
Adds to the kdoc, detailing how the builder instance of `CompilerPipeline` is not thread safe.

Possible TODOs:
- Should we also document this for the `CompilerPipeline` interface (not just the builder instance)?
- Should we add annotations of non-thread safety?
- Should we also document that `EvaluatingCompiler` is not thread safe?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/src/org/partiql/lang/CompilerPipeline.kt [hunk 1]
```diff
@@ -93,7 +93,10 @@ interface CompilerPipeline  {
             builder(valueFactory).build()
     }
 
-    /** An implementation of the builder pattern for instances of [CompilerPipeline]. */
+    /**
+     * An implementation of the builder pattern for instances of [CompilerPipeline]. The created instance of
+     * [CompilerPipeline] is NOT thread safe and should NOT be used to compile queries concurrently.
+     */
```
Comment (dlurton) [inline lang/src/org/partiql/lang/CompilerPipeline.kt:99]: Add:

> If used in a multithreaded application, use one instance of [CompilerPipeline] per thread.

**Review by dlurton — CHANGES_REQUESTED** [review]
- Should we also document this for the CompilerPipeline interface (not just the builder instance)?
    - I would, yes.
- Should we add annotations of non-thread safety?
    - Of what annotations specifically do you speak of?  I am not aware of any that aren't part of another library and I'd rather avoid taking a dependency on a new library just for that.
- Should we also document that EvaluatingCompiler is not thread safe?
   - `EvaluatingCompiler` is `internal`, but you can add a quick note to it if you'd like.

File: lang/src/org/partiql/lang/CompilerPipeline.kt (hunk 1 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/CompilerPipeline.kt:104]: Added.

**Comment by alancai98** [comment]
New revision includes updated kdoc to `CompilerPipeline` builder and interface. Also adds a note about thread safety to `EvaluatingCompiler`.

Regarding
>Of what annotations specifically do you speak of? I am not aware of any that aren't part of another library and I'd rather avoid taking a dependency on a new library just for that.

None in particular. I wasn't sure if there was any standard annotation to denote a class was not thread safe.

**Review by dlurton — APPROVED** [review]

**Pull request merged**